File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.java
import java.util.*;

class Solution {
    public int solution(int n, int[] stations, int w) {
        
        int answer = 0;
        int st = 1;
        int range = w*2+1;
        
        for(int i=0; i<stations.length; i++){
            int rangeLeft = stations[i] - w -1;
            int len = rangeLeft - st + 1;
            if(len > 0){
                answer += Math.ceil((double)len/range);
            }
            st = stations[i] + w + 1;
        }
        
        if(st < n){
            int len = n-st;
            answer += Math.ceil((double)len/range);
        }else if(st == n){
            answer += 1;
        }
        
        return answer;
    }
}

File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.js
function solution(n, stations, w) {
   
    var answer = 0;
    var st = 1;
    var range = w*2+1;
    
    for(let i=0; i<stations.length; i++){
        var leftRange = stations[i] - w - 1;
        var len = leftRange - st + 1;
        answer += Math.ceil(len/range);
        st = stations[i] + w + 1;
    }
    
    if(st < n){
        answer += Math.ceil((n-st)/range);
    }else if(st == n){
        answer += 1
    }
    
    return answer;
}